external-builds/pytorch/build_prod_wheels.py

```diff
@@ -553,6 +553,46 @@ def do_build_triton(
     return f"{triton_wheel_name}=={installed_triton_version}"
 
 
+def copy_msvc_libomp_to_torch_lib(pytorch_dir: Path):
+    # When USE_OPENMP is set (it is by default), torch_cpu.dll depends on OpenMP.
+    #
+    # Typically implementations of OpenMP are:
+    #   * Intel OpenMP, `libiomp`, which PyTorch upstream uses
+    #   * MSVC OpenMP, `libomp140`, which we'll use here since we have MSVC already
+    #   * (?) LLVM OpenMP (https://openmp.llvm.org/)?
+    #
+    # Torch's CMake build selects which OpenMP to use in `FindOpenMP.cmake`,
+    # then the relevant .dll files must be copied into the torch/lib/ folder or
+    # torch will fail to initialize. This feels like something that could be
+    # handled upstream as part of the centralized setup.py and/or CMake build
+    # processes, but given the varied scripts and build workflows upstream and
+    # multiple choices for where to source an implementation, we handle it here.
+    #
+    # If we wanted to switch to Intel OpenMP, we could:
+    #   1. Install Intel OpenMP (and MKL?)
+    #   2. Set CMAKE_INCLUDE_PATH and CMAKE_LIBRARY_PATH (?) so `FindOpenMP.cmake` finds them
+    #   3. Copy `libiomp5md.dll` to torch/lib
+    # Then remove the rest of the code from this function.
+
+    vc_tools_redist_dir = os.environ.get("VCToolsRedistDir", "")
+    if not vc_tools_redist_dir:
+        print("Warning: VCToolsRedistDir not set, can't copy libomp to torch lib")
```

Comment:
should we throw an error instead of a warning? If a user hasn't installed MSVC and/or set `VCToolsRedistDir`, they'll probably face problems at runtime anyway.

Reply:
Sure, I can switch this to an error. There may be ways to get the build working without this environment variable set, but yeah - there would probably be runtime errors later if this is skipped.

Reply:
Good point. I suppose it's a question of which is the less painful thing for the user to do - install MSVC or manually copy the DLL from somewhere.

Reply:
For developers running this script (building from source), MSVC is a requirement. The "developer command prompt" / vcvarsall / etc. ways to set up MSVC that I'm aware of _do_ set this environment variable. There are many other environment variables set also though, and it might be possible to build without this _particular_ variable set. 

For example, we use a different variable here: https://github.com/ROCm/TheRock/blob/6282bd46135b480d563077e32b40af291cd6931d/build_tools/github_actions/build_configure.py#L32-L39